### PR TITLE
qa/tests: fix test_list_enabled_modules timeout error

### PR DESCRIPTION
**Pull request by nizamial09**
This test deals with enabling/disabling the modules. The assumption I have is after enabling the
module, test will wait for an active mgr where dashboard runs but its not able to find it in time and it fails. so taking inspiration from https://github.com/ceph/ceph/pull/58995/commits/6c7253be6f6fbfa6faed7a539cb78847fec04580 adding retries and logs to see if that's the case

Fixes: https://tracker.ceph.com/issues/62972

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test api

File: qa/tasks/mgr/dashboard/test_mgr_module.py [hunk 1]
```diff
@@ -52,6 +53,7 @@ def test_list_disabled_module(self):
         self.assertIsNotNone(module_info)
         self.assertFalse(module_info['enabled'])
 
+    @retry(on_exception=RuntimeError, tries=2, delay=0.5, logger=log)
```
Comment (epuertat) [inline qa/tasks/mgr/dashboard/test_mgr_module.py:56]: Why not using the logger from this file?
```suggestion
    @retry(on_exception=RuntimeError, tries=2, delay=0.5, logger=logger)
```

File: qa/tasks/mgr/dashboard/test_mgr_module.py [hunk 2]
```diff
@@ -52,6 +53,7 @@ def test_list_disabled_module(self):
...
     def test_list_enabled_module(self):
         self._ceph_cmd(['mgr', 'module', 'enable', 'iostat'])
         self.wait_until_rest_api_accessible()
```
Comment (epuertat) [inline qa/tasks/mgr/dashboard/test_mgr_module.py:59]: Shouldn't this command be enough to make it work? Maybe the fix needs to happen there?

**Review by epuertat — APPROVED** [review]
LGTM! Just a few observations!

File: qa/tasks/mgr/dashboard/test_mgr_module.py (hunk 1 above)
Comment (nizamial09) [reply qa/tasks/mgr/dashboard/test_mgr_module.py]: thanks, fixed it

File: qa/tasks/mgr/dashboard/test_mgr_module.py (hunk 2 above)
Comment (nizamial09) [reply qa/tasks/mgr/dashboard/test_mgr_module.py]: i moved the retry to `wait_until_rest_api_accessible`

**Comment by nizamial09** [comment]
I've retried the api tests several times but i wasn't able to reproduce the mgr modules issue. I'll retrigger it couple of times again (making sure last push didn't break anything) and if all goes well will merge this.

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
merging this since even after a lot of retries i am not seeing the issue anymore.

**Pull request merged**